hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/SqlCreateMapping.java

```diff
@@ -163,7 +170,61 @@ public void unparse(SqlWriter writer, int leftPrec, int rightPrec) {
         }
     }
 
-    private void printIndent(SqlWriter writer) {
+    public static String unparse(Mapping mapping) {
+        SqlPrettyWriter writer = new SqlPrettyWriter(SqlPrettyWriter.config());
+
+        writer.keyword("CREATE MAPPING");
+        writer.print(escape(mapping.name()));
+
+        if (mapping.externalName() != null) {
+            writer.print(" ");
+            writer.keyword("EXTERNAL NAME");
+            writer.print(escape(mapping.externalName()));
+        }
+
+        List<MappingField> fields = mapping.fields();
+        if (fields.size() > 0) {
```

Comment:
Generating empty list would result in syntax error. Not including any fields means exactly that fields should be resolved.

Reply:
That's the point, the empty field list must only come from an error. So in case of erroneous input we'll produce erroneous output, not an output that means something else.